pkg/instrumentation/auto/annotation.go

```diff
@@ -37,29 +37,38 @@ type AnnotationMutators struct {
 	daemonSetMutators   map[string]instrumentation.AnnotationMutator
 	statefulSetMutators map[string]instrumentation.AnnotationMutator
 	defaultMutator      instrumentation.AnnotationMutator
+	injectAnnotations   map[string]struct{}
 }
 
 // RestartNamespace sets the restartedAtAnnotation for each of the namespace's supported resources and patches them.
-func (m *AnnotationMutators) RestartNamespace(ctx context.Context, namespace *corev1.Namespace) {
+func (m *AnnotationMutators) RestartNamespace(ctx context.Context, namespace *corev1.Namespace, mutatedAnnotations map[string]string) {
 	restartAndPatchFunc := m.patchFunc(ctx, setRestartAnnotation)
```

Comment:
nit: Move chain callback declaration to here
```suggestion
	restartAndPatchFunc := chainCallbacks(m.shouldRestartFunc(mutatedAnnotations), m.patchFunc(ctx, setRestartAnnotation))
```